azkaban-common/src/main/resources/host_name_alias.perl

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env perl
+
+# A script to get the VIP name of the current host.
+
+use strict;
+use warnings;
+
+# Get a list of IP addresses that the host has.
+sub get_ip_addrs() {
+    my @ip_addrs = ();
+    # Use the absolute path of ifconfig
+    # since the script may be invoked without PATH being set.
+    for my $line (`/sbin/ifconfig -a`) {
```

Comment:
"ifconfig" is being replaced by the "ip" command in linux distributions.  This will not work on newer debian or redhat/centos 7 distros. 

Also instead of using backticks, you want to use the system function for executing external commands, ("perldoc -f system"). System this will perform a fork and wait for the child in the background.  Additionally you can right shift $? by 8 to get the exit code of the command called by system. 

Another ifconfig (or ip) command replacement suggestion would be to read /proc/net/tcp for IPv4 and convert the hex strings to IPs and /proc/net/if_inet6 for IPv6 addresses